src/utils.jl

```diff
@@ -428,3 +428,33 @@ Get the amount dimension of a quantity (e.g., mol^(uamount)).
 """
 uamount(q::UnionAbstractQuantity) = uamount(dimension(q))
 uamount(d::AbstractDimensions) = d.amount
+
+"""
+Utility functions for `DynamicQuantitiesCollection`
+"""
+
+for f in (:enumerate, :length, :lastindex)
+    @eval begin
+        Base.$f(u::DynamicQuantitiesCollection) = $f(u.collection)
+    end
+end
+
+Base.getindex(u::DynamicQuantitiesCollection, i::Union{Int,  Symbol}) = getindex(u.collection, i)
+Base.getindex(u::DynamicQuantitiesCollection, i::UInt16) = getindex(u.collection, convert(Int, i))
+
+Base.setindex!(u::DynamicQuantities.DynamicQuantitiesCollection{Dict{Symbol, T}}, i::Int64, s::Symbol) where T <: Integer = setindex!(u.collection, i, s)
+Base.get(u::DynamicQuantitiesCollection{Dict{Symbol, INDEX_TYPE}}, a, b) = get(u.collection, a, b)
+
+Base.iterate(u::DynamicQuantitiesCollection) = iterate(u.collection)
+Base.iterate(u::DynamicQuantitiesCollection, i::Int) = iterate(u.collection, i)
+
+Base.intersect(u::DynamicQuantitiesCollection, v::AbstractSet) = intersect(u.collection, v)
+Base.intersect(v::AbstractSet, u::DynamicQuantitiesCollection) = intersect(v, u.collection)
```

Comment:
I don’t think these are used anywhere FYI

Reply:
Used here: https://github.com/SymbolicML/DynamicQuantities.jl/blob/main/src/symbolic_dimensions.jl#L5.

Maybe even Constant-collection can use the same struct; but I don't want to overload this PR.

Reply:
Oh I see, sorry. I’m not sure why the testing coverage didn’t catch that.